app/src/androidTest/java/ch/hikemate/app/navigation/HikeMateAppNavigationTest.kt

```diff
@@ -1,67 +1,112 @@
+    // Navigate to the sign in screen
+    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_EMAIL).performClick()
+    composeTestRule.onNodeWithTag(Screen.SIGN_IN_WITH_EMAIL).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(SignInWithEmailScreen.TEST_TAG_GO_TO_SIGN_UP_BUTTON)
+        .performClick()
+    composeTestRule.onNodeWithTag(Screen.CREATE_ACCOUNT).assertIsDisplayed()
+
+    // Navigate back to the sign in screen
+    composeTestRule.onNodeWithTag(BackButton.BACK_BUTTON_TEST_TAG).performClick()
+    composeTestRule.onNodeWithTag(Screen.SIGN_IN_WITH_EMAIL).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(BackButton.BACK_BUTTON_TEST_TAG).performClick()
+    composeTestRule.onNodeWithTag(Screen.AUTH).assertIsDisplayed()
   }
 
-  // @Test
-  fun testNavigationToPlannedHikesScreen() {
-    composeTestRule.setContent { HikeMateApp() }
+  /**
+   * Tests the whole navigation flow from the auth screen to the map screen. All the screens are
+   * tested because signing in takes time.
+   */
+  @Test
+  fun testWholeNavigation() {
+    // Create a user in order to skip the auth screen
+    setupUser()
+
     composeTestRule.onNodeWithTag(Screen.AUTH).assertIsDisplayed()
 
-    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_GOOGLE).performClick()
+    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_EMAIL).performClick()
+    composeTestRule.onNodeWithTag(Screen.SIGN_IN_WITH_EMAIL).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(SignInWithEmailScreen.TEST_TAG_EMAIL_INPUT)
+        .performTextInput(email)
+    composeTestRule
+        .onNodeWithTag(SignInWithEmailScreen.TEST_TAG_PASSWORD_INPUT)
+        .performTextInput(password)
+    composeTestRule.onNodeWithTag(SignInWithEmailScreen.TEST_TAG_SIGN_IN_BUTTON).performClick()
+
+    // Wait for the map to load
+    Thread.sleep(1000)
+
     composeTestRule.onNodeWithTag(Screen.MAP).assertIsDisplayed()
 
-    // Open the sidebar
+    // Go to planned hikes
     composeTestRule.onNodeWithTag(TEST_TAG_SIDEBAR_BUTTON).performClick()
-
     composeTestRule.onNodeWithTag(TEST_TAG_DRAWER_ITEM_PREFIX + Route.SAVED_HIKES).performClick()
-
     composeTestRule.onNodeWithTag(TEST_TAG_SAVED_HIKES_SECTION_CONTAINER).assertIsDisplayed()
-  }
 
-  // @Test
-  fun testNavigationToProfileScreen() {
-    composeTestRule.setContent { HikeMateApp() }
-    composeTestRule.onNodeWithTag(Screen.AUTH).assertIsDisplayed()
-
-    composeTestRule.onNodeWithTag(SignInScreen.TEST_TAG_SIGN_IN_WITH_GOOGLE).performClick()
-    composeTestRule.onNodeWithTag(Screen.MAP).assertIsDisplayed()
-    // Open the sidebar
+    // Go to profile screen
     composeTestRule.onNodeWithTag(TEST_TAG_SIDEBAR_BUTTON).performClick()
     composeTestRule.onNodeWithTag(TEST_TAG_DRAWER_ITEM_PREFIX + Route.PROFILE).performClick()
     composeTestRule.onNodeWithTag(Screen.PROFILE).assertIsDisplayed()
+
+    // Delete the user
+    deleteUser()
```

Comment:
Warning: If the test fails, the `deleteUser` is not called. You must find a way to call it even when an assert fails

Reply:
Yes you're right!